### PR TITLE
[MM-49641] - System console setting to disable channel autocomplete

### LINKED ISSUE #22068: System console setting to disable channel autocomplete
In some locales, the `~` character is used heavily in written communication. 

This ticket proposes the addition of a system console config setting that disables the channel autocomplete to avoid conflicts.

In the System Console > Posts section, add a new config setting: 

**Enable channel autocomplete**      < x ] true    [ x > false
When enabled, users can type the `</sub>` character to autocomplete channel names when composing messages. When disables, autocomplete suggestions for channel names will not be displayed. 

*____*

Functional notes:
- When disabled, the `~` character should not trigger auto-complete pop-ups when users are composing messages on web or mobile (may need a separate ticket). 

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-49641

### PR DESCRIPTION
**Pull request by gitstart**
### Summary
System console setting to disable channel autocomplete

### Demo video
https://www.loom.com/share/5b23a1494f234cc6a538c4af1aec873b

Fixes https://github.com/mattermost/mattermost-server/issues/22068

#### Release note
```release-note
Adds system console setting to enable/disable channel autocomplete
```
---
This code was written and reviewed by GitStart Community. Growing future engineers, one PR at a time.

### CODE REVIEW

File: components/autosize_textarea.tsx [hunk 1]
```diff
@@ -108,6 +108,7 @@ export class AutosizeTextarea extends React.PureComponent<Props> {
         Reflect.deleteProperty(props, 'providers');
         Reflect.deleteProperty(props, 'channelId');
         Reflect.deleteProperty(props, 'forwardedRef');
+        Reflect.deleteProperty(props, 'onWidthChange');
```
Comment (gitstart) [inline components/autosize_textarea.tsx:111]: `onWidthChange` was getting passed to textarea

Screenshot 2023-02-16 at 19 11 41

File: packages/types/src/config.ts [hunk 2]
```diff
@@ -724,6 +725,7 @@ export type ExperimentalSettings = {
     EnableRemoteClusterService: boolean;
     EnableAppBar: boolean;
     PatchPluginsReactDOM: boolean;
+    EnableChannelAutocomplete: boolean;
```
Comment (gitstart) [inline packages/types/src/config.ts:728]: Placed under experimental settings following a review from the API PR https://github.com/mattermost/mattermost-server/pull/22353#discussion_r1112078333

**Review by neallred — APPROVED** [review]
Verified locally that this and the accompanying back-end PR work together to implement the enabling and disabling autocomplete for channels. I was able to enable and disable from both the system console and from the config file, and the enable/disable worked both for main compose message window and for threads message window.

**Review by hmhealey — APPROVED** [review]
Thanks for the PR! This looks good to me.

I did notice that we aren't adding/removing the `ChannelMentionProvider` dynamically based on the value of the `Textbox`'s `channelAutocompleteEnabled` prop, but I don't think that's something we need to worry about since it behaves the same as the `supportsCommands` prop

**Comment by collinewait** [comment]
@hmhealey these changes have been moved to the monorepo in this PR https://github.com/mattermost/mattermost-server/pull/22353 cc @AGMETEOR , @M-ZubairAhmed

**Comment by hmhealey** [comment]
Thanks! I'm going to close this one then

**Pull request closed without merging**